core/org.wso2.carbon.user.core/src/main/java/org/wso2/carbon/user/core/hybrid/HybridRoleV2Manager.java

```diff
@@ -0,0 +1,1251 @@
+package org.wso2.carbon.user.core.hybrid;
+
+import org.apache.commons.collections.CollectionUtils;
+import org.apache.commons.lang.ArrayUtils;
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.wso2.carbon.CarbonConstants;
+import org.wso2.carbon.user.api.RealmConfiguration;
+import org.wso2.carbon.user.core.UserCoreConstants;
+import org.wso2.carbon.user.core.UserRealm;
+import org.wso2.carbon.user.core.UserStoreException;
+import org.wso2.carbon.user.core.authorization.AuthorizationCache;
+import org.wso2.carbon.user.core.common.UserRolesCache;
+import org.wso2.carbon.user.core.constants.UserCoreDBConstants;
+import org.wso2.carbon.user.core.constants.UserCoreErrorConstants;
+import org.wso2.carbon.user.core.jdbc.JDBCUserStoreManager;
+import org.wso2.carbon.user.core.jdbc.caseinsensitive.JDBCCaseInsensitiveConstants;
+import org.wso2.carbon.user.core.util.DatabaseUtil;
+import org.wso2.carbon.user.core.util.UserCoreUtil;
+import org.wso2.carbon.utils.dbcreator.DatabaseCreator;
+import org.wso2.carbon.utils.xml.StringUtils;
+
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.UUID;
+import java.util.regex.Matcher;
+import javax.sql.DataSource;
+
+import static org.wso2.carbon.user.core.constants.UserCoreErrorConstants.ErrorMessages.ERROR_CODE_DUPLICATE_WHILE_WRITING_TO_DATABASE;
+import static org.wso2.carbon.user.core.hybrid.HybridJDBCConstants.COUNT_INTERNAL_ONLY_ROLES_V2_SQL_WITH_AUDIENCE;
+import static org.wso2.carbon.user.core.hybrid.HybridJDBCConstants.COUNT_INTERNAL_ROLES_V2_SQL_WITH_AUDIENCE;
+
+public class HybridRoleV2Manager implements HybridRoleBasedManager {
+
+    private static Log log = LogFactory.getLog(JDBCUserStoreManager.class);
+    private final int DEFAULT_MAX_ROLE_LIST_SIZE = 1000;
+    private final int DEFAULT_MAX_SEARCH_TIME = 1000;
+    protected UserRealm userRealm = null;
+    protected UserRolesCache userRolesCache = null;
+    int tenantId;
+
+    private String audience = "organization";
+    private String audienceId;
+    private DataSource dataSource;
+    private RealmConfiguration realmConfig;
+    private String isCascadeDeleteEnabled;
+    private boolean userRolesCacheEnabled = true;
+    private static final String APPLICATION_DOMAIN = "Application";
+    private static final String WORKFLOW_DOMAIN = "Workflow";
+
```

Comment:
https://checkstyle.sourceforge.io/apidocs/com/puppycrawl/tools/checkstyle/checks/coding/DeclarationOrderCheck.html#:~:text=According%20to%20Code%20Conventions%20for,Instance%20variables.

check the order